### ISSUE TITLE
Ability to specify kill signal on step definition

### ISSUE THREAD

**Issue by fishnux** [yohamta/dagu#275]
Hi,

In a DAG step, I'm running an app that supports graceful shutdown upon CTRL+C. I experimented stopping the DAG in the middle of execution, but it seems it's not the same thing as doing CTRL+C.

As a feature request, can we define the TERM signal in the step definition when an exit is requested? So, for this specific step, I can use SIGINT (CTRL+C)

Thanks 🙂 

Edit: Meanwhile, I edited the source-code of the app to trap SIGTERM as well

**Comment by yohamta** [comment]
Thanks for the future request! :)

Currently if a signal sent to `dagu` process it will send the same signal to the running step. So creating other DAG to stop signal may be the workaround for now.

example:
```yaml
schedule: "0 18 * * *" # schedule to stop the main DAG
steps:
  - name: stop the DAG
    command: pkill -INT -f "dagu start <DAG name>"
```

I was thinking about alternative solutions (more generic ones) to deal with the situation.
- Add `--signal=<signal name>` option to `stop` command (ex: `dagu stop --signal=2 some_app.yaml`)
- Add `signalOnStop` field to step field
- Add some handler function that enables to do something before forwarding signal by dagu

I think the first one is the one we can add easily for now, if it fits your situation. How do you think?

**Comment by fishnux** [comment]
Hi,

I use the UI to stop the DAG. The `signalOnStop` field to the step field would make more sense in the scenario I think. If this field is not specified then fallback to --signal or the signal passed to dagu stop

**Comment by yohamta** [comment]
I see :) Then adding `signalOnStop` as a field of `Step` sounds good to me.

example:
```yaml
steps:
  step:
    - name: step 1
      command: app.py
      signalOnStop: 2
```

**Comment by yohamta** [comment]
@fishnux This feature is implemented in `v1.7.7`. Thank you!
